### ISSUE TITLE
Device does not support TouchID error

### ISSUE THREAD

**Issue by unkownymouse** [naoufal/react-native-touch-id#112]
I'm getting `Device does not support TouchID` error when I ran `TouchID.isSupported()`

```
componentWillMount() {
        //Determine Biometric Type
        TouchID.isSupported()
            .then(biometryType => {
                if (biometryType === 'FaceID') {
                    console.log('FaceID');
                } else {
                    console.log('TouchID');
                }
            })
            .catch(error => {
                // Failure code
                console.log(error);
            });
}
```

I'm getting this error message in the console:

<img width="783" alt="screen shot 2018-04-07 at 8 57 58 am" src="https://user-images.githubusercontent.com/32857628/38450811-d9a96e8a-3a41-11e8-9c57-51f402469c26.png">

---
## System Info:

"react": "16.3.1",
"react-native": "0.54.4",
"react-native-touch-id": "^4.0.1",

macOS High Sierra: 10.13.4
Xcode: Version 9.3 (9E145) with Swift 4.0.3

---
## Tested on:
iPhone X (Physical device)
iPhone X (Simulator)
iPhone 8 (Simulator)

**Comment by frmrgr** [comment]
I had the same issue because I had my simulator left open from xcode 9.0, so it had ios 11.0. I quit simulator app and launched it back from new xcode (in my case 9.0.1), this time it had ios 11.0.1, and face id worked.

**Comment by unkownymouse** [comment]
@frmrgr I also tried that. I restarted iPhone Simulator and again started it from Terminal(Mostly I launch the simulator from Terminal). 

My simulator and physical device iOS is 11.3. I also tried to delete `node_modules` and `npm install` then `react-native link`. Then I deleted the app from device and again installed it freshly. Even after this, it is showing the same error.

**Comment by muresanandrei** [comment]
I've also got this error does anyone know a fix ?

**Comment by rdev** [comment]
When you try wrong fingerprints a bunch of times on lock screen, it disables the sensor and asks you to use a passcode.

Turns out this also works if you fail authenticate inside the apps
And when that happens, until you lock your phone and unlock with a passcode, it returns `RCTTouchIDNotSupported`.

Lock your phone, unlock with a passcode and relaunch your app.

Got me for about 20 minutes -_-

**Comment by antbaobao** [comment]
has anyone  solved this  this problem ?

**Comment by CusProjects** [comment]
I had the same problem but was because i hadn't registered a fingerprint yet.

**Comment by JslinSir** [comment]
When device fingerprint identification is supported, if the "RCTTouchID NotSupported" message is reported, we can consider the number of times exceeded the limit

**Comment by chengsam** [comment]
@rdev You're right, but the error is quite misleading, hope the error can be logged more clearly.

**Comment by jpellissari** [comment]
Hello guys, I had the same error message. It was a simple step that is not listed anywhere. I'm using xcode 10.1 and simulator with iOS 12.1

Open the simulator and go to the menu "Hardware/Face ID(Touch ID)", there is an option called "enrolled", make sure this option is checked and try again!
